src/models/subgraph-entities/participant.ts

```diff
@@ -3,17 +3,15 @@ import { BigNumber } from 'ethers'
 export type Participant = Partial<{
```

Comment:
What's the reasoning behind turning these subgraph entity types into interfaces with some optional args vs. using Partial<>?

For the record i don't feel too strongly about using types over interfaces or vice versa, but the important bit to note is that these entities aren't always returned with all properties. These represent the "full" schema, but in a graph query only the properties specified in the `opts.keys` are returned.

it's been a back-of-mind task for a while to find a tighter system for this so if you're onto something better than i'm down to use it, just trying to make sure i'm on your level.

Reply:
@peripheralist Yeah, I've been considering what to do here exactly, as I see the difficulty. 

I didn't think about `opts.keys` for optional keys, I was only looking at the parseJson for each respective DTO. I could try to find a better TypeScript solution... Maybe using `Pick<>` with a dynamic second argument.

I'll try a few things but in the meantime, it'll be prudent to revert these changes back to exclusively `Partial` types.

Reply:
Cool. Using some type strategy that depends on `keys` would be cool, but I wouldn't consider it essential for this PR. Unless you feel like diving into it now I'm happy to just come back to it later